x/merkledb/trieview.go

```diff
@@ -231,6 +222,8 @@ func (t *trieView) calculateNodeIDs(ctx context.Context) error {
 		}
 		defer t.nodesAlreadyCalculated.Set(true)
 
+		oldRoot := maybe.Bind(t.root, (*node).clone)
```

Comment:
I don't think this needs to be cloned since before values don't get edited

Reply:
I think we need to clone here because `t.root` can change during the `for key, change := range t.changes.values` loop. We can't just do `oldRoot := t.root` because we'll end up with `oldRoot` being set to the new root.